File: nysol_python/mcmd/methods/convTex/db/smp/func_cat.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='cat'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,str1,str2,str3
1,abc,def,ghi
2,A,,CDE
3,,,
4,,,XY
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
3つの項目str1,str2,str3を ``"#"`` の区切り文字を挿入して併合する。
'''
script['sh_code']='''
mcal c='cat("#",$s{str1},$s{str2},$s{str3})' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='cat("#",$s{str1},$s{str2},$s{str3})', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='tokenが空文字の場合'
script['comment']='''
'''
script['sh_code']='''
mcal c='cat("",$s{str1},$s{str2},$s{str3})' a=rsl i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcal(c='cat("",$s{str1},$s{str2},$s{str3})', a='rsl', i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='ワイルドカードを利用した例'
script['comment']='''
``str`` から始まる項目( ``str1,str2,str3`` )をワイルドカード「 ``str*`` 」によって指定している。
'''
script['sh_code']='''
mcal c='cat("",$s{str*})' a=rsl i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcal(c='cat("",$s{str*})', a='rsl', i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_cat.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str1,str2,str3
1,abc,def,ghi
2,A,,CDE
3,,,
4,,,XY
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Concatenate 3 columns \verb|str1,str2,str3| with the inclusion of \verb|"#"| as delimiter token in between characters.
EOF
scp=<<'EOF'
more dat1.csv
mcal c='cat("#",$s{str1},$s{str2},$s{str3})' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Empty token"
comment=<<'EOF'
EOF
scp=<<'EOF'
mcal c='cat("",$s{str1},$s{str2},$s{str3})' a=rsl i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)


############## Example 3
title="Example using wildcard"
comment=<<'EOF'
Use wildcard to specify columns names that start with \verb|str| (\verb|str1,str2,str3|) such as \verb|str*|.
EOF
scp=<<'EOF'
mcal c='cat("",$s{str*})' a=rsl i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

